### PR TITLE
[AMD] Merge Dockerfiles for ROCm

### PR DESCRIPTION
**Pull request by akao-amd**
## Motivation

`rocm720.Dockerfile` and `rocm.Dockerfile` shares a lot in common. It is time to unify them.

## Modifications

Making `rocm720.Dockerfile` a superset of `rocm.Dockerfile` was the design choice back in #17799 . This PR mostly replaces `rocm.Dockerfile` with `rocm720.Dockerfile`, and modifies all the users of it accordingly.  Refactoring workflow files is a reasonable next step but not include in this PR. 

## Accuracy Tests

Both `rocm700-mi35x` and `rocm720-mi35x` pass the accuracy test of DEEPSEEK-R1-MXFP4-Preview model.

## Benchmarking and Profiling

N/A.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: docker/rocm.Dockerfile [hunk 1]
```diff
@@ -81,6 +106,21 @@ USER root
...
+      *rocm720*) \
+        echo "ROCm 7.2 flavor detected from GPU_ARCH=${GPU_ARCH}"; \
+        cd /opt/rocm/share/amd_smi \
+        && python3 -m pip install --no-cache-dir . \
+        ;; \
+      *) \
+        echo "Not rocm720 (GPU_ARCH=${GPU_ARCH}), skip amdsmi installation"; \
+        ;; \
+    esac
+RUN
```
[the remark this answers is not in the document]
Comment (akao-amd) [reply docker/rocm.Dockerfile]: No, it does serve a purpose, as pointing out in https://github.com/sgl-project/sglang/pull/19091

File: docker/rocm.Dockerfile [hunk 2]
```diff
@@ -375,9 +424,95 @@ RUN /bin/bash -lc 'set -euo pipefail; \
...
+    csv.writer(f).writerows(rows)
+
+# 4) Re-zip as a wheel
+with zipfile.ZipFile(out_whl, "w", compression=zipfile.ZIP_DEFLATED) as z:
+    for p in work.rglob("*"):
+        if p.is_file():
+            z.write(p, p.relative_to(work).as_posix())
+
+print("Wrote", out_whl)
+PY
```
[the remark this answers is not in the document]
Comment (akao-amd) [reply docker/rocm.Dockerfile:467]: This was a pure hot fix for specific ROCm 7.2 environment.  It is ugly, but we don't expect to maintain this in a long term.

File: docker/rocm.Dockerfile [hunk 3]
```diff
@@ -115,18 +172,8 @@ RUN cd aiter \
...
-RUN if [ "$BUILD_TRITON" = "1" ]; then \
-        pip uninstall -y triton \
-     && git clone ${TRITON_REPO} \
-     && cd triton \
-     && git checkout ${TRITON_COMMIT} \
-     && cd python \
-     && python setup.py install; \
-    fi
+        fi \
+     && echo "export PYTHONPATH=/sgl-workspace/aiter:\${PYTHONPATH}" >> /etc/bash.bashrc
```
[the remark this answers is not in the document]
Comment (akao-amd) [reply docker/rocm.Dockerfile]: Solved.

File: docker/rocm.Dockerfile (hunk 1 above)
Comment (HaiShaw) [inline docker/rocm.Dockerfile]: Please make SMI installation only to ROCm 7.2

File: docker/rocm.Dockerfile [hunk 4]
```diff
@@ -47,8 +72,8 @@ ARG SGL_BRANCH=${SGL_DEFAULT}
 # Version override for setuptools_scm (used in nightly builds)
 ARG SETUPTOOLS_SCM_PRETEND_VERSION=""
 
-ARG TRITON_REPO="https://github.com/ROCm/triton.git"
-ARG TRITON_COMMIT="improve_fa_decode_3.0.0"
+ARG TRITON_REPO="https://github.com/triton-lang/triton.git"
+ARG TRITON_COMMIT="42270451990532c67e69d753fbd026f28fcc4840"
```
Comment (HaiShaw) [inline docker/rocm.Dockerfile:79]: This changes Triton for ROCm 7.0 image, haven't gone through QA and perf bench.
Should do separate Triton version for now as from separate docker files.

File: docker/rocm.Dockerfile [hunk 5]
```diff
@@ -102,11 +142,28 @@ RUN if [ "$BUILD_LLVM" = "1" ]; then \
 # (SETUPTOOLS_SCM_PRETEND_VERSION is set later for SGLang nightly builds and would otherwise
 # leak into AITER's version when AITER uses setuptools_scm)
 ENV SETUPTOOLS_SCM_PRETEND_VERSION=
-RUN pip uninstall -y aiter
+RUN pip uninstall -y aiter \
+ && pip install psutil pybind11 # Required by AITER setup.py
```
Comment (HaiShaw) [inline docker/rocm.Dockerfile]: `-y` here, pay attention to duplicate install or run into different version

File: docker/rocm.Dockerfile [hunk 6]
```diff
@@ -102,11 +142,28 @@ RUN if [ "$BUILD_LLVM" = "1" ]; then \
...
+      *rocm720*) \
+        echo "ROCm 7.2 flavor detected from GPU_ARCH=${GPU_ARCH}"; \
+        cd aiter \
+        && sed -i '459 s/if.*:/if False:/' aiter/ops/triton/attention/pa_mqa_logits.py; \
+        ;; \
+      *) \
+        echo "Not rocm720 (GPU_ARCH=${GPU_ARCH}), skip patch"; \
+        ;; \
+    esac
+
```
Comment (HaiShaw) [inline docker/rocm.Dockerfile:168]: Make above ROCm 7.2 only

File: docker/rocm.Dockerfile [hunk 7]
```diff
@@ -117,17 +174,6 @@ RUN cd aiter \
           sh -c "GPU_ARCHS=$GPU_ARCH_LIST python setup.py develop"; \
         fi
```
Comment (HaiShaw) [inline docker/rocm.Dockerfile:179]: Lost `&& echo "export PYTHONPATH=/sgl-workspace/aiter:\${PYTHONPATH}" >> /etc/bash.bashrc` in case ROCm 7.2
Please be careful at merging two dockerfiles

File: docker/rocm.Dockerfile [hunk 8]
```diff
@@ -117,17 +174,6 @@ RUN cd aiter \
...
-# Triton (TODO: remove this after Triton is no longer needed)
-RUN if [ "$BUILD_TRITON" = "1" ]; then \
-        pip uninstall -y triton \
-     && git clone ${TRITON_REPO} \
-     && cd triton \
-     && git checkout ${TRITON_COMMIT} \
-     && cd python \
-     && python setup.py install; \
-    fi
-
```
Comment (HaiShaw) [inline docker/rocm.Dockerfile:130]: Should keep this for ROCm 7.0

File: docker/rocm.Dockerfile
Comment (HaiShaw) [inline docker/rocm.Dockerfile]: Consider to remove vLLM build

File: docker/rocm.Dockerfile [hunk 9]
```diff
@@ -375,9 +423,95 @@ RUN /bin/bash -lc 'set -euo pipefail; \
...
+RUN if [ "$BUILD_TRITON" = "1" ]; then \
+        pip uninstall -y triton \
+     && apt install -y cmake \
+     && git clone ${TRITON_REPO} triton-custom \
+     && cd triton-custom \
+     && git checkout ${TRITON_COMMIT} \
+     && pip install -r python/requirements.txt \
+     && pip install -e .; \
+    fi
+
```
Comment (HaiShaw) [inline docker/rocm.Dockerfile:498]: L426-L509, pls keep it identical as in ROCm 7.2 dockerfile, and apply it only to ROCm7.2 build

**Review by HaiShaw — CHANGES_REQUESTED** [review]
Let's keep using `release-docker-amd-nightly.yml` vs. `release-docker-amd-rocm720-nightly.yml`

File: docker/rocm.Dockerfile (hunk 1 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile]: Yes, they are indeed ROCm 7.2 only.  It is a conditional block that tells if a build is for ROCm 7.2 (`rocm720` substring) or otherwise.

File: docker/rocm.Dockerfile (hunk 4 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile:79]: Both 700 images have `BUILD_TRITON=0`.  I don't think there is any concerns here.

File: docker/rocm.Dockerfile (hunk 5 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile]: Thanks for the catch.  Will fix later.

File: docker/rocm.Dockerfile (hunk 6 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile:168]: Yes, they are indeed ROCm 7.2 only. It is a conditional block that tells if a build is for ROCm 7.2 (rocm720 substring) or otherwise.

File: docker/rocm.Dockerfile (hunk 7 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile:179]: We (YC, BingXu) discussed yesterday that the line wasn't needed.

File: docker/rocm.Dockerfile (hunk 8 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile:130]: It was just move to bottom, not removed.

File: docker/rocm.Dockerfile
Comment (akao-amd) [reply docker/rocm.Dockerfile]: Will remove them later.

File: docker/rocm.Dockerfile (hunk 9 above)
Comment (akao-amd) [reply docker/rocm.Dockerfile:498]: No.  I believe `BUILD_TRITON` is already a better flag for this block and serves its purpose correctly. 

Currently ROCm 7.2 images gives `BUILD_TRITON=1` while ROCm 7.0 images give `BUILD_TRITON=0`.  The flag has a **finer granularity** that can serve us well.  For example, if we don't want to update gfx942-rocm720 for some reason, then it can keep as is while we may have fixed the triton dependency issue for gfx950-rocm720.

File: docker/rocm.Dockerfile (hunk 7 above)
Comment (HaiShaw) [reply docker/rocm.Dockerfile:179]: This is needed so `sys.path` on ROCm 7.2 is set properly to include `/sgl-workspace/aiter`

Comment (akao-amd) [reply docker/rocm.Dockerfile:179]: Sorry but can you elaborate?  If it is the case, why doesn't ROCm 7.0 image need it in the first place?
CC. @yctseng0211 @bingxche

**Comment by bingxche** [comment]
Nightly Release successfully
rocm700: https://github.com/sgl-project/sglang/actions/runs/22471253280/job/65088525837
rocm720: https://github.com/sgl-project/sglang/actions/runs/22471255572/job/65088532429

**Review by HaiShaw — APPROVED** [review]
